networkx/generators/tests/test_directed.py

```diff
@@ -119,6 +150,11 @@ def test_with_replacement(self):
         G = random_uniform_k_out_graph(n, k, with_replacement=True)
         assert G.is_multigraph()
         assert all(d == k for v, d in G.out_degree())
+        n = 10
+        k = 9
+        G = random_uniform_k_out_graph(n, k, with_replacement=False, self_loops=False)
+        assert not nx.number_of_selfloops(G)
```

Comment:
Being explicit here is both easier to understand (IMO) and avoids unexpected results from relying on the "truthiness" of non-Boolean objects.

```suggestion
        assert nx.number_of_selfloops(G) == 0
```